pkg/dump/dump.go

```diff
@@ -142,3 +160,174 @@ func getTimescaleInfo(dbURI string) (util.TsInfo, error) {
+	defer wg.Done()
+
+	movedJobs := make(map[int64]time.Time)
+	runningJobs := true
+
+	ticker := time.NewTicker(30 * time.Second)
+	conn, err := util.GetDBConn(context.Background(), dbURI)
+	if err != nil {
+		compressionJobMoverWarn(err)
+		return
+	}
+	defer conn.Close(context.Background())
+	var tsMajorVersion int
+	err = conn.QueryRow(context.Background(), "SELECT split_part(extversion, '.', 1)::INT FROM pg_catalog.pg_extension WHERE extname='timescaledb' LIMIT 1").Scan(&tsMajorVersion)
+	if err != nil {
+		compressionJobMoverWarn(err)
+		return
+	}
+	runningJobsSQL, moveSQL, replaceJobSQL, err := compressionJobMoverSQL(tsMajorVersion, pauseUDAs)
+	if err != nil {
+		compressionJobMoverWarn(err)
+		return
+	}
+	var runningJobIDs string
+	for {
+		rows, err := conn.Query(context.Background(), moveSQL)
+		if err != nil {
+			compressionJobMoverWarn(err)
+			return
+		}
+		var (
+			jobID     int64
+			origStart time.Time
+		)
+		for rows.Next() {
+			if err = rows.Scan(&jobID, &origStart); err != nil {
+				compressionJobMoverWarn(err)
+				return
+			}
+			if _, ok := movedJobs[jobID]; !ok {
+				if verbose {
+					fmt.Printf("%sMoved Job: %d\n", time.Now().Format("2006/01/02 15:04:05 "), jobID)
+				}
+				movedJobs[jobID] = origStart
+			}
+		}
+		rows.Close()
+		if runningJobs {
+
+			err = conn.QueryRow(context.Background(), runningJobsSQL).Scan(&runningJobIDs)
+			if err != nil && err != pgx.ErrNoRows {
+				compressionJobMoverWarn(err)
+				return
+			}
+			if err == pgx.ErrNoRows {
+				err = nil
+				runningJobs = false
+				fmt.Printf("%sJobs: %s have stopped, continuing \n ", time.Now().Format("2006/01/02 15:04:05 "), runningJobIDs)
+				jobsStopped <- true // signal back to our parent that it's safe to move on, the jobs have stopped
+			}
+			if runningJobs {
+				if verbose {
+					fmt.Printf("%sBackgroud jobs %s are currently running\n these jobs may cause the dump to fail by causing deadlocks, the main dump will wait for them to finish \n ", time.Now().Format("2006/01/02 15:04:05 "), runningJobIDs)
+				}
+			}
+		}
+		select {
+		case <-cleanup:
+			var jobMoved bool
+			for jobID, origStart := range movedJobs {
+				if verbose {
+					fmt.Printf("%sReplacing Job: %d\n", time.Now().Format("2006/01/02 15:04:05 "), jobID)
+				}
+				err = conn.QueryRow(context.Background(), replaceJobSQL, jobID, origStart).Scan(&jobMoved)
+				if err != nil && err != pgx.ErrNoRows {
+					compressionJobMoverWarn(err)
+					return
+				}
+			}
+			return
+		case <-ticker.C:
+			continue
+		}
+	}
+}
+
+func compressionJobMoverWarn(err error) {
+	fmt.Println(time.Now().Format("2006/01/02 15:04:05 ")+"WARNING: ", fmt.Errorf("problem while rescheduling jobs: %w", err))
+}
+
+func compressionJobMoverStop(wg *sync.WaitGroup, cleanup chan bool) {
+	cleanup <- true
+	wg.Wait()
+}
+
+func compressionJobMoverSQL(tsMajorVersion int, pauseUDAs bool) (runningJobsSQL string, moveSQL string, replaceJobSQL string, err error) {
```

Comment:
consider splitting this out into 3 functions after the other functions are split --- not sure here, use your judgment

Reply:
Didn't do this as it seemed like then I'd just have to pass more state around and I think the SQL is all related to each other, so this seemed cleaner to me.